lib/rules/selector-no-deprecated/index.mjs

```diff
@@ -0,0 +1,167 @@
+import selectorParser from 'postcss-selector-parser';
+const { isPseudo, isPseudoElement, isTag } = selectorParser;
+
+import {
+	deprecatedHtmlTypeSelectors,
+	deprecatedPseudoClasses,
+	deprecatedPseudoElements,
+	deprecatedSvgTypeSelectors,
+} from '../../reference/selectors.mjs';
+import { isRegExp, isString } from '../../utils/validateTypes.mjs';
+import getRuleSelector from '../../utils/getRuleSelector.mjs';
+import isCustomSelector from '../../utils/isCustomSelector.mjs';
+import isStandardSyntaxRule from '../../utils/isStandardSyntaxRule.mjs';
+import isStandardSyntaxSelector from '../../utils/isStandardSyntaxSelector.mjs';
+import optionsMatches from '../../utils/optionsMatches.mjs';
+import parseSelector from '../../utils/parseSelector.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import uniteSets from '../../utils/uniteSets.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'selector-no-deprecated';
+
+const messages = ruleMessages(ruleName, {
+	expected: (unfixed, fixed) => `Expected "${unfixed}" to be "${fixed}"`,
+	rejected: (selector) => `Unexpected deprecated selector "${selector}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/selector-no-deprecated',
+	fixable: true,
+};
+
+/** @type {Partial<Record<string, string | null>>} */
+const DEPRECATED_PSEUDO_CLASSES_REMAP = {
+	'focus-ring': 'focus-visible',
+	matches: 'is',
+};
+
+const DEPRECATED_PSEUDO_PATTERN = new RegExp(
+	`${[...deprecatedPseudoClasses, ...deprecatedPseudoElements].join('|')}`,
+	'i',
+);
+
+/** @type {ReadonlySet<string>} */
+const DEPRECATED_SELECTORS = uniteSets(deprecatedHtmlTypeSelectors, deprecatedSvgTypeSelectors);
+
+const DEPRECATED_SELECTOR_PATTERN = new RegExp(`${[...DEPRECATED_SELECTORS].join('|')}`, 'i');
+
+/** @type {import('stylelint').CoreRules[ruleName]} */
+const rule = (primary, secondaryOptions) => {
+	return (root, result) => {
+		const validOptions = validateOptions(
+			result,
+			ruleName,
+			{ actual: primary },
+			{
+				actual: secondaryOptions,
+				possible: {
+					ignoreSelectors: [isString, isRegExp],
+				},
+				optional: true,
+			},
+		);
+
+		if (!validOptions) return;
+
+		root.walkRules((ruleNode) => {
+			if (!isStandardSyntaxRule(ruleNode)) return;
+
+			const selector = getRuleSelector(ruleNode);
+
+			const hasDeprecatedPseudo = DEPRECATED_PSEUDO_PATTERN.test(selector);
+			const hasDeprecatedSelector = DEPRECATED_SELECTOR_PATTERN.test(selector);
+
+			if (!hasDeprecatedPseudo && !hasDeprecatedSelector) return;
+
+			const resolvedRoot = parseSelector(selector, result, ruleNode);
+
+			if (!resolvedRoot) return;
+
+			resolvedRoot.walk((node) => {
+				if (hasDeprecatedPseudo && isPseudo(node)) {
```

Comment:
Could you add more test cases ?

e.g.
`:is(), :matches() {}` // don't assume the order
`::backdrop {}` // no false positives due to `:drop`

Reply:
Added